### This Notbook is my personal learning journey with ChatGPT

### 📌 Step 3: Variables (Very Important!)

Variables store information.

#### Exercise 2

Create variables for:

* Your name
* Your age
* Your favorite number

Then print them.

In [7]:
# Exercise 2
name = "M Abid"
age = 25
fav_num = 27

print(name)
print(age)
print(fav_num)

M Abid
25
27


Excellent work, Abid! 👏🔥

You did everything correctly.

#### Exercise 2 Review

```python
name = "M Abid"
age = 25
fav_number = 27

print(name)
print(age)
print(fav_number)
```

✔ Variables created correctly
✔ Correct data types (string + integers)
✔ Correct printing

Very good start

### 🚀 Step 4: Numbers & Math in Python

Python can do math like a calculator.

## Basic Operators

| Symbol | Meaning        |
| ------ | -------------- |
| `+`    | Addition       |
| `-`    | Subtraction    |
| `*`    | Multiplication |
| `/`    | Division       |
| `**`   | Power          |
| `%`    | Remainder      |


#### 🧠 Bonus Challenge (Important for problem solving)

Try this:

```python
num = 7
```

Print:

* Double of the number
* Triple of the number
* Square of the number

In [12]:
# Exercise 3
x = 15
y = 4

print(x + y)
print(x - y)
print(x * y)
print(x / y)
print(x % y)

19
11
60
3.75
3


In [9]:
# Bonus Challenge
num = 7

print(num * 2)
print(num * 3)
print(num ** 2)

14
21
49


#### Exercise 3 Review

```python
x = 15
y = 4
print(x + y)
print(x - y)
print(x * y)
print(x / y)
print(x % y)
```

✔ Addition → 19
✔ Subtraction → 11
✔ Multiplication → 60
✔ Division → 3.75
✔ Remainder → 3

Perfect 👌

#### Bonus Review

```python
num = 7
print(num * 2)
print(num * 3)
print(num ** 2)
```

✔ Double → 14
✔ Triple → 21
✔ Square → 49

Excellent thinking

### 🚀 Step 5: User Input (Now it gets interesting!)

Now your program can **talk to the user**.

⚠ Important:
`input()` always gives you TEXT (string).

If you want a number, you must convert it:


#### Exercise 4

1. Ask the user for their name.
2. Ask the user for their age.
3. Print:

   ```
   Hello Abid, next year you will be 26
   ```

(Hint: You need `int()` for age.)

#### 🎯 Mini Challenge (Think a little!)

Make a small program that:

* Asks for two numbers
* Prints their sum

Example output:

```
Enter first number: 5
Enter second number: 3
The sum is 8
```

In [13]:
# Exercise 4
name = input("What is your name? ")
age = int(input("What is your age? "))

print("Hello", name, "next year you will be", age + 1)

Hello M Abid next year you will be 26


In [15]:
# Mini Challenge
first_number = int(input("Enter first number: "))
second_number = int(input("Enter second number: "))
total = first_number + second_number

print("The sum is:", total)

The sum is: 8


Let’s review carefully:

#### Exercise 4 Review

```python
name = input("What is your name? ")
age = int(input("Enter you age: "))
print("Hello", name, "next year you will be", age + 1)
```

🟢 Very good:

* Correct use of `input()`
* Correct use of `int()`
* Correct printing

#### Mini Challenge Review

```python
first_number = int(input("Enter first number: "))
second_number = int(input("Enter second number: "))
total = first_number + second_number
print("The sum is ", total)
```

✔ Perfect logic
✔ Correct conversion
✔ Correct addition